add errors in the conn to the form

tarzan opened the pull request:
When creating a form with the conn instead of with a changeset struct, there is no easy way of displaying error messages.

This is an attempt to make that easier by allowing `:errors` as an option to `form_for/4`.
In the controller:
```ex
def my_post_method(conn, params) do
  case do_stuff(params) do
    {:ok, object} -> render(conn, view_template)
    {:error, reason} -> render(conn, view_template, errors: reason)
  end
end
```
and in the template
```eex
<%= form_for @conn, "/do_stuff", [errors: @conn.assigns[:errors]], fn f -> do %>
  <%= text_input f, :field %>
  <%= error_tag f, :field %>
<% end %>
```
where `error_tag` is the custom function as mentioned in the [migration guides](https://gist.github.com/chrismccord/557ef22e2a03a7992624).

chrismccord commented:
I think in this case it's better to handle the errors yourself, for example you could simply use the `@errors` assign yourself in the template rather than lifting it off the form.

josevalim commented:
The advantage of lifting of the form is that the error tag helper that we have just works but i am not sure how comfortable i am with hijacking the assigns errors key. Maybe we should rather pass errors as an explicit option on `form_for`?

josevalim commented:
By passing an option maybe we get the best of both worlds. Unsure.

chrismccord commented:
Ah good point. I handle thought about error_tag. I really like the idea of passing the errors as an option to form_for! That would get my vote.

tarzan commented:
Thanks for the swift response(s)! I'll rework this as an option for `form_for`.

tarzan commented:
Hey @chrismccord and @josevalim I changed it to allowing `:errors` as an option to `form_for/4`. 

Added a bit of documentation as well. Hope you like it.

Cheers!

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
